autoload/SpaceVim/plugins/manager.vim

```diff
@@ -566,14 +587,43 @@ function! s:msg_on_build_failed(name, ...) abort
   endif
 endfunction
 
+" - foo.vim: Updating failed.
+function! s:add_to_failed_list(name) abort
+  if index(s:failed_plugins, a:name) < 0
+    call add(s:failed_plugins, a:name)
+  endif
+endfunction
+
+function! s:reinstall_update_failed() abort
+  if len(s:failed_plugins) > 0 && s:retry_cnt > 0
+    " close plugin manager
+    let s:on_reinstall = 1
+    call SpaceVim#logger#warn(' [ plug manager ] Reinstalling Failed Plugins. Remaining Retries: '.s:retry_cnt)
+    call SpaceVim#plugins#manager#update(s:failed_plugins)
+    let s:failed_plugins = []
+    let s:retry_cnt -= 1
+  endif
+endfunction
+
 function! s:new_window() abort
-  if s:plugin_manager_buffer != 0 && bufexists(s:plugin_manager_buffer)
+  if s:plugin_manager_buffer != 0 && bufexists(s:plugin_manager_buffer) && s:on_reinstall == 0
     " buffer exist, process has not finished!
     return 0
-  elseif s:plugin_manager_buffer != 0 && !bufexists(s:plugin_manager_buffer)
+  elseif s:plugin_manager_buffer != 0 && !bufexists(s:plugin_manager_buffer) && s:on_reinstall == 0
     " buffer is hidden, process has not finished!
     call s:resume_window()
     return 1
+  elseif s:plugin_manager_buffer != 0 && bufexists(s:plugin_manager_buffer) && s:on_reinstall == 1
+    call setbufvar(s:plugin_manager_buffer, '&ma', 1)
+    let current_nr = winnr()
+    let winnr = bufwinnr(s:plugin_manager_buffer)
+    if winnr > -1
+      exe winnr . 'wincmd w'
+      silent exe '1,$d'
```

Comment:
`normal! gg"_dG` should be better, as it will not change the content in register.

Reply:
Got it. Fixed now.